table/block_based/filter_policy_internal.h

```diff
@@ -127,6 +137,10 @@ class BloomFilterPolicy : public FilterPolicy {
   // behavior with format_version < 5 just in case.)
   int whole_bits_per_key_;
 
+  // For configuring Ribbon filter: a desired value for 1/fp_rate. For
+  // example, 100 -> 1% fp rate.
+  double desired_one_in_fp_rate_;
```

Comment:
My hypothesis is that it's easier to configure things when you can use whole numbers for most common choices, with pretty good granularity. Think powers of 2 or powers of 10 up to about 1000. Beyond roughly 1000, numbers start to get uncomfortable to read and write, and those settings should not be needed for most applications anyway.

And I don't like specifying things as percentages. When someone writes 0.01, did they mean 1% or 1 in 10,000?

I'm reluctant to emphasize bits per key because that can have a complicated relationship with FP rate now that we are moving beyond Bloom and Ribbon128 does not scale infinitely in that way as Bloom does. But maybe people do care about that, in terms of how many bits of memory do I need to dedicate for (more or less) each key in my database.

In any case, this part is relatively hidden and can be changed / updated.

Reply:
at least for public interface, we don't have `_one_in_`, but already has some `_ratio` like: https://github.com/facebook/rocksdb/blob/bcba372352644c76267f4da8c0eb373682a17472/include/rocksdb/advanced_options.h#L298-L306